src/Solnet.Rpc/SolanaRpcBatchComposer.cs

```diff
@@ -0,0 +1,384 @@
+        }
+
+        /// <summary>
+        /// Returns a batch of JSON RPC requests
+        /// </summary>
+        /// <returns></returns>
+        public JsonRpcBatchRequest CreateJsonRequests()
+        {
+            var reqs = new JsonRpcBatchRequest();
+            reqs.AddRange(_reqs.Select(x => x.Req));
+            return reqs;
+        }
+
+        /// <summary>
+        /// Execute a batch request and process the response into the expected native types.
+        /// Batch failure execption will invoke callbacks with an exception.
+        /// </summary>
+        /// <param name="client"></param>
+        public JsonRpcBatchResponse Execute(IRpcClient client)
+        {
+            return ExecuteAsync(client).Result;
+        }
+
+        /// <summary>
+        /// Execute a batch request and process the response into the expected native types.
+        /// Batch failure execption will invoke callbacks with an exception.
+        /// </summary>
+        /// <param name="client"></param>
+        public async Task<JsonRpcBatchResponse> ExecuteAsync(IRpcClient client)
+        {
+            var reqs = this.CreateJsonRequests();
+            var response = await client.SendBatchRequestAsync(reqs);
+            if (response.WasSuccessful)
+                return ProcessBatchResponse(response);
+            else
+                return ProcessBatchFailure(response);
+        }
+
+        /// <summary>
+        /// Execute a batch request and process the response into the expected native types.
+        /// Batch failure execption will throw an Exception and bypass callbacks.
+        /// </summary>
+        /// <param name="client"></param>
+        public JsonRpcBatchResponse ExecuteWithFatalFailure(IRpcClient client)
+        {
+            return ExecuteWithFatalFailureAsync(client).Result;
+        }
+
+        /// <summary>
+        /// Execute a batch request and process the response into the expected native types.
+        /// Batch failure execption will throw an Exception and bypass callbacks.
+        /// </summary>
+        /// <param name="client"></param>
+        public async Task<JsonRpcBatchResponse> ExecuteWithFatalFailureAsync(IRpcClient client)
+        {
+            var reqs = this.CreateJsonRequests();
+            var response = await client.SendBatchRequestAsync(reqs);
+            if (response.WasSuccessful)
+                return ProcessBatchResponse(response);
+            else
+                throw new ApplicationException($"Batch was unsuccessful: {response.Reason}");
+        }
+
+        /// <summary>
+        /// Handles the conversion of the generic JSON deserialized response objects to the native types.
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        public JsonRpcBatchResponse ProcessBatchResponse(RequestResult<JsonRpcBatchResponse> response)
+        {
+            // TODO - make sure result was successful
```

Comment:
Do we need further validation? If not delete comment.

Reply:
Removed and method changed to be internal as calling logic and flow deals with the success/failure outcome.